src/JWT.ts

```diff
@@ -244,8 +251,40 @@ export function decodeJWT(jwt: string): JWTDecoded {
   if (!jwt) throw new Error('invalid_argument: no JWT passed into decodeJWT')
   try {
     const jws = decodeJWS(jwt)
-    const decodedJwt: JWTDecoded = Object.assign(jws, { payload: JSON.parse(decodeBase64url(jws.payload)) })
-    return decodedJwt
+    if (jws.header.cty === "JWT") {
+      // This is a multi-signature JWT using the nested JWT structure
+      // See point 5 of https://www.rfc-editor.org/rfc/rfc7519#section-7.1
+      const signature: GeneralJWSSignature[] = [];
+      const data = jws.data;
+      let i = 0;
+
+      let loopJws: JWSDecoded = {} as any;
+      Object.assign(loopJws, jws)
+
+      while (i < 200 && loopJws.header.cty === "JWT") {
```

Comment:
why 200 ?

Reply:
i suggest using a recursive function instead of loop

Reply:
I initially tried this and failed. While loop was easier for me to debug and reason about.

200 is arbitrary, just to prevent an infinite loop.